docs/src/ci.md

```diff
@@ -0,0 +1,167 @@
+
+!!! note
+
+    You don't need to add a new commit to set custom URLs. You can modify the commit message with `git commit --amend` and force push to the branch. This also starts the CI pipeline again.
+
+There is a last job, which checks if lines starting with `CI_UNIT_PKG_URL_` exist in the commit message of the latest commit. If so, the unit test will fail. This is required, because otherwise the merged code would depend on non merged changes in sub-packages and would be non-compatible.
+
+!!! note
+
+If you use `CI_UNIT_PKG_URL_`, the CI pipeline will fail which does not mean that the actual tests are failing.
+
+# Integration Tests for CI Users
+
+The integration tests are automatically triggered if you open a pull request that targets the `main` or `dev` branch, and the unit tests have passed. The integration tests itself are in an extra stage of the CI.
+
+![CI pipeline with unit and integration tests](CI_pipeline.png)
+
+If the tests pass successfully, you don't need to do anything. If the fails, i.e. the change breaks the interoperability with another package in the `QED.jl` ecosystem, the pull request will suspend, and one has two options to proceed:
+
+1. One can solve the problem locally, by changing the code of the modified (sub-) package. The workflow is the same as for fixing unit tests.
+2. One needs to modify the depending package, which failed in the integration test. In the following, we describe how to provide the necessary changes to the downstream package and make the CI pass the integration tests, which will resume the pull request.
+
+For better understanding, the package currently modified by the pull request is called `orig`, and the package that depends on it is referred to as `dep`. First, one should fork the package `dep` and checkout a new feature branch on this fork. The fix of the integration issue for `dep` is now developed on the feature branch. Once finished, the changes to `dep` are push to GitHub, and a pull request on `dep` is opened to check in the changes. By default, the unit test for `dep` should fail, because the CI in `dep` needs to use the modified version of `orig`. The solution for this problem is explained in the section [Unit Test for CI Users](#Unit-Test-for-CI-Users). Using this, one should develop the fix on the feature branch until the CI of `dep` passes all unit tests. In this case, the original pull request in the upstream package `orig` can be resumed. Therefore, one needs to tell the CI of `orig` that the integration tests should use the fixed version package `dep`, which is still on the feature branch in a pull request on `dep`. In order to proceed, the CI on `orig` needs information on where the fix for `dep` is located. This is given to the CI of `orig` in a commit message on the origin branch of the pull request on `orig`, one just needs to add a new line with the following format to the commit message:
```

Comment:
I added a sentence, where I described how the the `Project.toml` looks, for better understanding.

Be aware, it line 129 we use `user` instead of `dep`. The documentation is written in the view of `orig`. You develop `orig`. Therefore you use the packages `dep` and your are used by the packages `user`.

Reply:
Can you please check, if the new sentences clarifies it.